### ISSUE TITLE
No client Vue plugin images with FireFox

### ISSUE THREAD

**Issue by amenak77** [freedomotic/fd-vue-webapp#158]
![firefox_prob](https://user-images.githubusercontent.com/13914556/44620959-1545df80-a89e-11e8-864d-de14079740da.jpg)

A little comparison between Chrome and FireFox about plugin panel with online client demo; no plugin images with FireFox (OS win 7 64 bit). May we consider this OS obsolete and ignore the issue?

**Comment by amenak77** [comment]
#113 No issue with Opera browser

**Comment by amenak77** [comment]
Why does only FireFox rise a CORS exception? It's very strange...

**Comment by P3trur0** [comment]
Why don't we consider to setup a Snapshot testing mechanism to verify if the GUI layout is properly working on specific browser?

**Comment by amenak77** [comment]
What do you mean?

**Comment by amenak77** [comment]
@p3trur0 please take a look to freedomotic issue #280 if my POW makes sense

**Comment by P3trur0** [comment]
@amenak77 I mean that these kind of tests could be performed automatically through a CI build.
We could setup a snapshot tests pipeline triggering build failures when the actual layout is not the one expected. See [here](https://jestjs.io/docs/en/snapshot-testing.html), for example.

**Comment by P3trur0** [comment]
@amenak77 indeed my proposal is about graphical layout issues, not about backend.

**Comment by amenak77** [comment]
@P3trur0 OK, now it's clear; it seems a very interesting approach. Are you familiar with this kind of test?

**Comment by P3trur0** [comment]
@amenak77 yes, I did few trials with **Jest**.

**Comment by amenak77** [comment]
@p3trur0 Good! Could you have a little time to create a model? It would be very useful to fight against the multi browser compatibility issue (at last!)

**Comment by amenak77** [comment]
I'm sorry, this issue is very similat to #113 opened by Mauro, but the real problem is: why firefox only does arise a CORS exception?

**Comment by amenak77** [comment]
It seems that there is a firefox addon to enable the user to use CORS everywhere by altering http responses

https://addons.mozilla.org/en-US/firefox/addon/cors-everywhere/?src=recommended

**Comment by amenak77** [comment]
![firefox_err_nocors](https://user-images.githubusercontent.com/13914556/44947946-1bdfd400-ae15-11e8-8b2c-52442f4ed78f.jpg)

I've installed firefox addon and no more CORS errors, but no plugin images, again. In console, there is one million error messages

**Comment by tranlehaiquan** [comment]
The problem both in chrome and firefox. Try to replace the images with real address it work (I think just problem went get with axios). I wonder why we need to fetch image with axios and parse it to data, instead of that do it with simple function:
```html
<img :src="getIconAddress(getPluginFromStore.uuid + '-' + getPluginFromStore.currentPluginStatus.toLowerCase() + '.png')">
```
```Javascript
getIconAddress: function (pluginIcon) {
        return `http://localhost:9111/v3/resources/${pluginIcon}`
      }
```
![image](https://user-images.githubusercontent.com/17347993/46469048-74cbcf00-c7fc-11e8-9cf5-e9d385a0fef1.png)

**Comment by landreussi** [comment]
I think icons is better than images in this case,for many reasons
would you thought about replace images to icons?

Personal suggestion: https://fontawesome.com/how-to-use/on-the-web/using-with/vuejs

**Comment by mcicolella** [comment]
@tranlehaiquan thanks for your suggestion. It fixed the issue.

**Comment by mcicolella** [comment]
@landreussi no we don't want to use icons. We share the images with plugins and other cilents (e.g. Jfrontend).

**Comment by mcicolella** [comment]
Fixed in 5e00d72d3f5e4fabcfcd226eee5568bf64336c5d
